lamindb/models/_describe.py

```diff
@@ -127,81 +127,113 @@ def format_bytes(bytes_value):
         return f"{bytes_value / (1024**4):.1f} TB"
 
 
-def describe_artifact_general(
-    self: Artifact,
+def describe_dataset(
+    obj: Artifact | Collection,
```

Comment:
I think you're reverting quite a bit of my change from a few days ago.

<img width="1207" height="153" alt="image" src="https://github.com/user-attachments/assets/1cb1909b-c958-4a3b-bc1a-1ba3137e7aa7" />

- https://github.com/laminlabs/lamindb/pull/2953

It used to accept both Artifact & Collection, but that led to all kinds of problems.

Rationale was: Given collection doesn't have its own metadata and in that sense is like every other sqlrecord I thought we need a proper solution for artifact, and then a generic solution for all other sqlrecords.

I'm looking through the code now but - for this reason - "philosophically" and operationally this doesn't seem great. 😅

Reply:
> I think you're reverting quite a bit of my change from a few days ago.

Evidently bad. After your general feedback, I'll look again to ensure that I don't do that.

Reply:
I went through all changes to `lamindb/models/_describe.py` and I think they're adding lots of complexity for little value (just `collection.describe()` doesn't bring us much) and at the cost of much danger: the performance aspect of the code will be very hard to keep under control and we might readily run into 8 sec `artifact.describe()` calls again.

So, I'd keep artifact isolated for now. I don't mind another function for collection that would hopefully leverage a smart abstraction re the rich formatting so that we don't duplicate everything.

Destination has to be `sqlrecord.describe()`, but I don't think many people care about this right now.